### ISSUE TITLE
LXD 3.19: Common start logic: Failed to change ACLs

### ISSUE THREAD

**Issue by Ne02ptzero** [lxc/lxd#6771]
All right, this is weird one:

I can't start an instance on an ARM server, the following is displayed:
```
$> lxc start LF2c8fa74f-4b5d-4512-b564-18242c727be2
Error: Common start logic: Failed to change ACLs on /var/lib/lxd/storage-pools/remote/containers/LF2c8fa74f-4b5d-4512-b564-18242c727be2/rootfs/var/log/journal
```

A strace on the lxd daemon responsible for starting the instance shows:
```
$> strace -ff lxd --debug 2>&1 | grep -i acl
[...]
[pid  1826] setxattr("/var/lib/lxd/storage-pools/remote/containers/LF2c8fa74f-4b5d-4512-b564-18242c727be2/rootfs/var/log/journal", "system.posix_acl_access", "\2\0\0\0\1\0\7\0\377\377\377\377\4\0\5\0\377\377\377\377\10\0\5\0\377\377\377\377\10\0\5", 52, 0) = -1 EINVAL (Invalid argument)
t=2020-01-24T19:45:52+0000 lvl=dbug msg="Failure for task operation: a19fab5f-4d72-44dd-8260-97637ea60194: Common start logic: Failed to change ACLs on /var/lib/lxd/storage-pools/remote/containers/LF2c8fa74f-4b5d-4512-b564-18242c727be2/rootfs/var/log/journal" 
```
(The `EINVAL` on the `setxattr` is _very_ weird, this errno value is not even documented)

The call is at `shared/idmap/shift_linux.go:290`

It is worth noting that I'm running a custom-compiled kernel on this machine, but the following options are enabled, so I don't think that is the problem:
```
$> cat .config | grep -i ACL
CONFIG_EXT4_FS_POSIX_ACL=y
CONFIG_REISERFS_FS_POSIX_ACL=y
CONFIG_JFS_POSIX_ACL=y
CONFIG_XFS_POSIX_ACL=y
CONFIG_BTRFS_FS_POSIX_ACL=y
CONFIG_F2FS_FS_POSIX_ACL=y
CONFIG_FS_POSIX_ACL=y
CONFIG_TMPFS_POSIX_ACL=y
CONFIG_NFS_V3_ACL=y
CONFIG_NFSD_V2_ACL=y
CONFIG_NFSD_V3_ACL=y
CONFIG_NFS_ACL_SUPPORT=y
CONFIG_CEPH_FS_POSIX_ACL=y
CONFIG_CIFS_ACL=y
CONFIG_9P_FS_POSIX_ACL=y
```

Here's an `ls` of the concerned file:
```
$> ls -la /var/lib/lxd/storage-pools/remote/containers/LF2c8fa74f-4b5d-4512-b564-18242c727be2/rootfs/var/log/journal
total 0
drwxr-sr-x+ 4 1000000 1000984  60 Jan 24 18:35 .
drwxr-xr-x  6 1000000 1000000 117 Jan 24 04:47 ..
drwxr-sr-x+ 2 root        984  28 Jan 24 18:35 2d4a27927566406caa8b52476c40ee3e
drwxr-sr-x  2 root        982   6 Dec  3 03:41 remote
```
Filesystem is XFS, with ceph rbd.

**Comment by Ne02ptzero** [comment]
I can see that the partiton is not mounted with `acl`:
```
/dev/rbd0 on /var/lib/lxd/storage-pools/remote/containers/LF2c8fa74f-4b5d-4512-b564-18242c727be2 type xfs (rw,relatime,attr2,discard,inode64,sunit=8192,swidth=8192,noquota)
```

**Comment by stgraber** [comment]
@Ne02ptzero is `acl` a required mount option for ACLs to work on xfs?
Also, this is the shifting code, so no ACL would be written if one wasn't already present, this is odd.

**Comment by Ne02ptzero** [comment]
> @Ne02ptzero is acl a required mount option for ACLs to work on xfs?

It appears that ACLs on XFS are enabled by default, as long they are enabled kernel-side.

**Comment by stgraber** [comment]
Ok, so that's not that.
What kernel version are you running? Not that it should really matter for ACLs...

Can you also maybe pull the existing ACL data from the file with `getfattr`?

**Comment by Ne02ptzero** [comment]
```
$> mount
[...]
/dev/rbd0 on /var/lib/lxd/storage-pools/remote/containers/LF2c8fa74f-4b5d-4512-b564-18242c727be2 type xfs (rw,relatime,attr2,inode64,sunit=8192,swidth=8192,noquota)
$> pwd
/var/lib/lxd/storage-pools/remote/containers/LF2c8fa74f-4b5d-4512-b564-18242c727be2/rootfs/var
$> getfattr -Rd log
# file: log/journal/2d4a27927566406caa8b52476c40ee3e/system.journal
user.crtime_usec=0sn521BeecBQA=
$> uname -a
Linux par1lf-cisco-prd-cl01pi01 4.19.97-v7+ #5 SMP Fri Jan 24 19:22:39 CET 2020 armv7l GNU/Linux
```

**Comment by stgraber** [comment]
@Ne02ptzero I've not had any luck reproducing this issue :(
Is it still affecting you?

If so, can you provide some more reproducing steps? Ideally starting from a commonly available image?

I have arm64 server hardware, so that part is covered.

**Comment by stgraber** [comment]
Also could you show:
```
getfacl -R /var/lib/lxd/storage-pools/remote/containers/LF2c8fa74f-4b5d-4512-b564-18242c727be2/rootfs/var/log/journal
```

**Comment by stgraber** [comment]
I believe we tracked this down and fixed it now

**Comment by ibrahimmenem** [comment]
Hi @stgraber is the fix included in the stable 4.1 snap package? I'm hitting a similar issue when setting `security.idmap.isolated true` 

> cat test.yml
```
architecture: x86_64
config:
  user.user-data: |-
    #cloud-config
    hostname: test.domain
    fqdn: test.domain
    debug:
      verbose: true
  security.idmap.isolated: true
devices:
  mnt:
    path: /mnt
    source: /mnt/test
    type: disk
  root:
    path: /
    pool: lxd
    size: 20GB
    type: disk
ephemeral: false
```

> lxc init  test-template test < test.yml

The test-template was created from vanilla ubuntu18.04 then installing some extra packages, the original container has   `security.idmap.isolated: true` set as well then `lxc publish` and  
`lxc image export`

> lxc start test
```
Error: Common start logic: Failed to change ACLs on /var/snap/lxd/common/lxd/storage-pools/lxd/containers/test/rootfs/var/log/journal
Try `lxc info --show-log test` for more info
```

> lxc info --show-log test
```
Name: test
Location: none
Remote: unix://
Architecture: x86_64
Created: 2020/06/11 09:38 UTC
Status: Stopped
Type: container
Profiles: default

Log:

```
> lxc info
```
config:
  images.auto_update_interval: "0"
api_extensions:
- storage_zfs_remove_snapshots
- container_host_shutdown_timeout
- container_stop_priority
- container_syscall_filtering
- auth_pki
- container_last_used_at
- etag
- patch
- usb_devices
- https_allowed_credentials
- image_compression_algorithm
- directory_manipulation
- container_cpu_time
- storage_zfs_use_refquota
- storage_lvm_mount_options
- network
- profile_usedby
- container_push
- container_exec_recording
- certificate_update
- container_exec_signal_handling
- gpu_devices
- container_image_properties
- migration_progress
- id_map
- network_firewall_filtering
- network_routes
- storage
- file_delete
- file_append
- network_dhcp_expiry
- storage_lvm_vg_rename
- storage_lvm_thinpool_rename
- network_vlan
- image_create_aliases
- container_stateless_copy
- container_only_migration
- storage_zfs_clone_copy
- unix_device_rename
- storage_lvm_use_thinpool
- storage_rsync_bwlimit
- network_vxlan_interface
- storage_btrfs_mount_options
- entity_description
- image_force_refresh
- storage_lvm_lv_resizing
- id_map_base
- file_symlinks
- container_push_target
- network_vlan_physical
- storage_images_delete
- container_edit_metadata
- container_snapshot_stateful_migration
- storage_driver_ceph
- storage_ceph_user_name
- resource_limits
- storage_volatile_initial_source
- storage_ceph_force_osd_reuse
- storage_block_filesystem_btrfs
- resources
- kernel_limits
- storage_api_volume_rename
- macaroon_authentication
- network_sriov
- console
- restrict_devlxd
- migration_pre_copy
- infiniband
- maas_network
- devlxd_events
- proxy
- network_dhcp_gateway
- file_get_symlink
- network_leases
- unix_device_hotplug
- storage_api_local_volume_handling
- operation_description
- clustering
- event_lifecycle
- storage_api_remote_volume_handling
- nvidia_runtime
- container_mount_propagation
- container_backup
- devlxd_images
- container_local_cross_pool_handling
- proxy_unix
- proxy_udp
- clustering_join
- proxy_tcp_udp_multi_port_handling
- network_state
- proxy_unix_dac_properties
- container_protection_delete
- unix_priv_drop
- pprof_http
- proxy_haproxy_protocol
- network_hwaddr
- proxy_nat
- network_nat_order
- container_full
- candid_authentication
- backup_compression
- candid_config
- nvidia_runtime_config
- storage_api_volume_snapshots
- storage_unmapped
- projects
- candid_config_key
- network_vxlan_ttl
- container_incremental_copy
- usb_optional_vendorid
- snapshot_scheduling
- container_copy_project
- clustering_server_address
- clustering_image_replication
- container_protection_shift
- snapshot_expiry
- container_backup_override_pool
- snapshot_expiry_creation
- network_leases_location
- resources_cpu_socket
- resources_gpu
- resources_numa
- kernel_features
- id_map_current
- event_location
- storage_api_remote_volume_snapshots
- network_nat_address
- container_nic_routes
- rbac
- cluster_internal_copy
- seccomp_notify
- lxc_features
- container_nic_ipvlan
- network_vlan_sriov
- storage_cephfs
- container_nic_ipfilter
- resources_v2
- container_exec_user_group_cwd
- container_syscall_intercept
- container_disk_shift
- storage_shifted
- resources_infiniband
- daemon_storage
- instances
- image_types
- resources_disk_sata
- clustering_roles
- images_expiry
- resources_network_firmware
- backup_compression_algorithm
- ceph_data_pool_name
- container_syscall_intercept_mount
- compression_squashfs
- container_raw_mount
- container_nic_routed
- container_syscall_intercept_mount_fuse
- container_disk_ceph
- virtual-machines
- image_profiles
- clustering_architecture
- resources_disk_id
- storage_lvm_stripes
- vm_boot_priority
- unix_hotplug_devices
- api_filtering
- instance_nic_network
- clustering_sizing
- firewall_driver
- projects_limits
- container_syscall_intercept_hugetlbfs
- limits_hugepages
- container_nic_routed_gateway
- projects_restrictions
- custom_volume_snapshot_expiry
- volume_snapshot_scheduling
- trust_ca_certificates
- snapshot_disk_usage
- clustering_edit_roles
- container_nic_routed_host_address
- container_nic_ipvlan_gateway
- resources_usb_pci
- resources_cpu_threads_numa
- resources_cpu_core_die
- api_os
- container_nic_routed_host_table
- container_nic_ipvlan_host_table
- container_nic_ipvlan_mode
- resources_system
- images_push_relay
api_status: stable
api_version: "1.0"
auth: trusted
public: false
auth_methods:
- tls
environment:
  addresses: []
  architectures:
  - x86_64
  - i686
  certificate: |
    -----BEGIN CERTIFICATE-----
    MIICdTCCAfqgAwIBAgIQBzAd20VxRS7Dj0jJ97hBZDAKBggqhkjOPQQDAzBZMRww

    hayT7OcsI7+f
    -----END CERTIFICATE-----
  certificate_fingerprint: 
XXXXXXX
  driver: lxc
  driver_version: 4.0.2
  firewall: xtables
  kernel: Linux
  kernel_architecture: x86_64
  kernel_features:
    netnsid_getifaddrs: "true"
    seccomp_listener: "true"
    seccomp_listener_continue: "true"
    shiftfs: "false"
    uevent_injection: "true"
    unpriv_fscaps: "true"
  kernel_version: 5.0.0-1026-gcp
  lxc_features:
    cgroup2: "true"
    mount_injection_file: "true"
    network_gateway_device_route: "true"
    network_ipvlan: "true"
    network_l2proxy: "true"
    network_phys_macvlan_mtu: "true"
    network_veth_router: "true"
    pidfd: "true"
    seccomp_notify: "true"
  os_name: Ubuntu
  os_version: "18.04"
  project: default
  server: lxd
  server_clustered: false
  server_name: name.domain
  server_pid: 17967
  server_version: "4.1"
  storage: zfs
  storage_version: 0.8.4-1~18.04.york0
```

**Comment by ibrahimmenem** [comment]
Doing some more search I found https://discuss.linuxcontainers.org/t/cannot-start-a-copied-container-failed-to-change-acls/7982/2 I'll remove `/var/log/journal` when creating the template to see if that solves the issue
